app/actions/configurations.py

```diff
@@ -0,0 +1,73 @@
+import pydantic
+
+from app.actions.core import AuthActionConfiguration, PullActionConfiguration
+from app.services.errors import ConfigurationNotFound
+from app.services.utils import find_config_for_action, FieldWithUIOptions, UIOptions, GlobalUISchemaOptions
+
+
+class AuthenticateConfig(AuthActionConfiguration):
+    subscription_key: pydantic.SecretStr = FieldWithUIOptions(
+        title="Ctrack Crystal Subscription Key",
+        description="A valid Ctrack Crystal API subscription key",
+        ui_options=UIOptions(
+            widget="password",
+        ),
+    )
+    username: str = FieldWithUIOptions(
+        title="Username",
+        description="The username for the Ctrack Crystal account",
+    )
+    password: pydantic.SecretStr = FieldWithUIOptions(
+        ...,
+        title="Password",
+        description="The password for the Ctrack Crystal account",
+        ui_options=UIOptions(
+            widget="password",
+        ),
+    )
+
+    ui_global_options: GlobalUISchemaOptions = GlobalUISchemaOptions(
+        order=[
+            "subscription_key",
+            "username",
+            "password"
+        ],
+    )
+
+
+class PullObservationsConfig(PullActionConfiguration):
+    pass
+
+
+class PullVehicleTripsConfig(PullActionConfiguration):
```

Comment:
If it's a subaction it should inherit from `InternalActionConfiguration`
```suggestion
class PullVehicleTripsConfig(InternalActionConfiguration):
```